Instance NODE_TYPES properties are missing in the typing of Node

The following code causes the typescript compiler to error:

```typescript
import { Node } from 'slimdom';

function test (node: Node) {
  return node.ELEMENT_NODE;
}
```

```
TSError: ⨯ Unable to compile TypeScript:
XXX: error TS2576: Property 'ELEMENT_NODE' is a static member of type 'Node'
```

The NODE_TYPE properties should be static as well as on the instance level

--- a/api/slimdom.api.json
+++ b/api/slimdom.api.json
@@ -6936,6 +6936,32 @@
             },
             {
               "kind": "Property",
+              "canonicalReference": "slimdom!Node#ATTRIBUTE_NODE:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "ATTRIBUTE_NODE: "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "ATTRIBUTE_NODE",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              },
+              "isStatic": false
+            },
+            {
+              "kind": "Property",
               "canonicalReference": "slimdom!Node.ATTRIBUTE_NODE:member",
               "docComment": "",
               "excerptTokens": [
@@ -6959,6 +6985,32 @@
                 "endIndex": 2
               },
               "isStatic": true
+            },
+            {
+              "kind": "Property",
+              "canonicalReference": "slimdom!Node#CDATA_SECTION_NODE:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "CDATA_SECTION_NODE: "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "CDATA_SECTION_NODE",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              },
+              "isStatic": false
             },
             {
               "kind": "Property",
@@ -7063,6 +7115,32 @@
             },
             {
               "kind": "Property",
+              "canonicalReference": "slimdom!Node#COMMENT_NODE:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "COMMENT_NODE: "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "COMMENT_NODE",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              },
+              "isStatic": false
+            },
+            {
+              "kind": "Property",
               "canonicalReference": "slimdom!Node.COMMENT_NODE:member",
               "docComment": "",
               "excerptTokens": [
@@ -7138,6 +7216,32 @@
             },
             {
               "kind": "Property",
+              "canonicalReference": "slimdom!Node#DOCUMENT_FRAGMENT_NODE:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "DOCUMENT_FRAGMENT_NODE: "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "DOCUMENT_FRAGMENT_NODE",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              },
+              "isStatic": false
+            },
+            {
+              "kind": "Property",
               "canonicalReference": "slimdom!Node.DOCUMENT_FRAGMENT_NODE:member",
               "docComment": "",
               "excerptTokens": [
@@ -7161,6 +7265,32 @@
                 "endIndex": 2
               },
               "isStatic": true
+            },
+            {
+              "kind": "Property",
+              "canonicalReference": "slimdom!Node#DOCUMENT_NODE:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "DOCUMENT_NODE: "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "DOCUMENT_NODE",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              },
+              "isStatic": false
             },
             {
               "kind": "Property",
@@ -7190,6 +7320,32 @@
             },
             {
               "kind": "Property",
+              "canonicalReference": "slimdom!Node#DOCUMENT_TYPE_NODE:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "DOCUMENT_TYPE_NODE: "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "DOCUMENT_TYPE_NODE",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              },
+              "isStatic": false
+            },
+            {
+              "kind": "Property",
               "canonicalReference": "slimdom!Node.DOCUMENT_TYPE_NODE:member",
               "docComment": "",
               "excerptTokens": [
@@ -7213,6 +7369,32 @@
                 "endIndex": 2
               },
               "isStatic": true
+            },
+            {
+              "kind": "Property",
+              "canonicalReference": "slimdom!Node#ELEMENT_NODE:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "ELEMENT_NODE: "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "ELEMENT_NODE",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              },
+              "isStatic": false
             },
             {
               "kind": "Property",
@@ -7242,6 +7424,32 @@
             },
             {
               "kind": "Property",
+              "canonicalReference": "slimdom!Node#ENTITY_NODE:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "ENTITY_NODE: "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "ENTITY_NODE",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              },
+              "isStatic": false
+            },
+            {
+              "kind": "Property",
               "canonicalReference": "slimdom!Node.ENTITY_NODE:member",
               "docComment": "",
               "excerptTokens": [
@@ -7265,6 +7473,32 @@
                 "endIndex": 2
               },
               "isStatic": true
+            },
+            {
+              "kind": "Property",
+              "canonicalReference": "slimdom!Node#ENTITY_REFERENCE_NODE:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "ENTITY_REFERENCE_NODE: "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "ENTITY_REFERENCE_NODE",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              },
+              "isStatic": false
             },
             {
               "kind": "Property",
@@ -7739,6 +7973,32 @@
             },
             {
               "kind": "Property",
+              "canonicalReference": "slimdom!Node#NOTATION_NODE:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "NOTATION_NODE: "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "NOTATION_NODE",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              },
+              "isStatic": false
+            },
+            {
+              "kind": "Property",
               "canonicalReference": "slimdom!Node.NOTATION_NODE:member",
               "docComment": "",
               "excerptTokens": [
@@ -7884,6 +8144,32 @@
               "propertyTypeTokenRange": {
                 "startIndex": 1,
                 "endIndex": 3
+              },
+              "isStatic": false
+            },
+            {
+              "kind": "Property",
+              "canonicalReference": "slimdom!Node#PROCESSING_INSTRUCTION_NODE:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "PROCESSING_INSTRUCTION_NODE: "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "PROCESSING_INSTRUCTION_NODE",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
               },
               "isStatic": false
             },
@@ -8060,6 +8346,32 @@
                 }
               ],
               "name": "replaceChild"
+            },
+            {
+              "kind": "Property",
+              "canonicalReference": "slimdom!Node#TEXT_NODE:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "TEXT_NODE: "
+                },
+                {
+                  "kind": "Content",
+                  "text": "number"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "releaseTag": "Public",
+              "name": "TEXT_NODE",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              },
+              "isStatic": false
             },
             {
               "kind": "Property",

--- a/api/slimdom.api.md
+++ b/api/slimdom.api.md
@@ -313,25 +313,43 @@ export abstract class Node {
     // (undocumented)
     static ATTRIBUTE_NODE: number;
     // (undocumented)
+    ATTRIBUTE_NODE: number;
+    // (undocumented)
     static CDATA_SECTION_NODE: number;
+    // (undocumented)
+    CDATA_SECTION_NODE: number;
     childNodes: Node[];
     cloneNode(deep?: boolean): this;
     // (undocumented)
     static COMMENT_NODE: number;
+    // (undocumented)
+    COMMENT_NODE: number;
     contains(other: Node | null): boolean;
     abstract _copy(document: Document): Node;
     // (undocumented)
     static DOCUMENT_FRAGMENT_NODE: number;
     // (undocumented)
+    DOCUMENT_FRAGMENT_NODE: number;
+    // (undocumented)
     static DOCUMENT_NODE: number;
+    // (undocumented)
+    DOCUMENT_NODE: number;
     // (undocumented)
     static DOCUMENT_TYPE_NODE: number;
     // (undocumented)
+    DOCUMENT_TYPE_NODE: number;
+    // (undocumented)
     static ELEMENT_NODE: number;
+    // (undocumented)
+    ELEMENT_NODE: number;
     // (undocumented)
     static ENTITY_NODE: number;
     // (undocumented)
+    ENTITY_NODE: number;
+    // (undocumented)
     static ENTITY_REFERENCE_NODE: number;
+    // (undocumented)
+    ENTITY_REFERENCE_NODE: number;
     firstChild: Node | null;
     hasChildNodes(): boolean;
     insertBefore<TNode extends Node>(node: TNode, child: Node | null): TNode;
@@ -347,18 +365,24 @@ export abstract class Node {
     normalize(): void;
     // (undocumented)
     static NOTATION_NODE: number;
+    // (undocumented)
+    NOTATION_NODE: number;
     ownerDocument: Document | null;
     readonly parentElement: Element | null;
     parentNode: Node | null;
     previousSibling: Node | null;
     // (undocumented)
     static PROCESSING_INSTRUCTION_NODE: number;
+    // (undocumented)
+    PROCESSING_INSTRUCTION_NODE: number;
     // Warning: (ae-forgotten-export) The symbol "RegisteredObservers" needs to be exported by the entry point index.d.ts
     _registeredObservers: RegisteredObservers;
     removeChild<TChild extends Node>(child: TChild): TChild;
     replaceChild<TChild extends Node>(node: Node, child: TChild): TChild;
     // (undocumented)
     static TEXT_NODE: number;
+    // (undocumented)
+    TEXT_NODE: number;
     abstract textContent: string | null;
 }
 

--- a/src/Node.ts
+++ b/src/Node.ts
@@ -22,6 +22,7 @@ import { asNullableObject, asNullableString, asObject } from './util/typeHelpers
  * @public
  */
 export default abstract class Node {
+	// Node types are exposed as properties of the constructor
 	static ELEMENT_NODE: number = NodeType.ELEMENT_NODE;
 	static ATTRIBUTE_NODE: number = NodeType.ATTRIBUTE_NODE;
 	static TEXT_NODE: number = NodeType.TEXT_NODE;
@@ -34,6 +35,20 @@ export default abstract class Node {
 	static DOCUMENT_TYPE_NODE: number = NodeType.DOCUMENT_TYPE_NODE;
 	static DOCUMENT_FRAGMENT_NODE: number = NodeType.DOCUMENT_FRAGMENT_NODE;
 	static NOTATION_NODE: number = NodeType.NOTATION_NODE; // historical
+
+	// Node types also exist as instance properties, assigned to the prototype below
+	public ELEMENT_NODE!: number;
+	public ATTRIBUTE_NODE!: number;
+	public TEXT_NODE!: number;
+	public CDATA_SECTION_NODE!: number;
+	public ENTITY_REFERENCE_NODE!: number; // historical
+	public ENTITY_NODE!: number; // historical
+	public PROCESSING_INSTRUCTION_NODE!: number;
+	public COMMENT_NODE!: number;
+	public DOCUMENT_NODE!: number;
+	public DOCUMENT_TYPE_NODE!: number;
+	public DOCUMENT_FRAGMENT_NODE!: number;
+	public NOTATION_NODE!: number; // historical
 
 	/**
 	 * Returns the type of node, represented by a number.
@@ -373,15 +388,15 @@ export default abstract class Node {
 	public abstract _copy(document: Document): Node;
 }
 
-(Node.prototype as any).ELEMENT_NODE = NodeType.ELEMENT_NODE;
-(Node.prototype as any).ATTRIBUTE_NODE = NodeType.ATTRIBUTE_NODE;
-(Node.prototype as any).TEXT_NODE = NodeType.TEXT_NODE;
-(Node.prototype as any).CDATA_SECTION_NODE = NodeType.CDATA_SECTION_NODE;
-(Node.prototype as any).ENTITY_REFERENCE_NODE = NodeType.ENTITY_REFERENCE_NODE; // historical
-(Node.prototype as any).ENTITY_NODE = NodeType.ENTITY_NODE; // historical
-(Node.prototype as any).PROCESSING_INSTRUCTION_NODE = NodeType.PROCESSING_INSTRUCTION_NODE;
-(Node.prototype as any).COMMENT_NODE = NodeType.COMMENT_NODE;
-(Node.prototype as any).DOCUMENT_NODE = NodeType.DOCUMENT_NODE;
-(Node.prototype as any).DOCUMENT_TYPE_NODE = NodeType.DOCUMENT_TYPE_NODE;
-(Node.prototype as any).DOCUMENT_FRAGMENT_NODE = NodeType.DOCUMENT_FRAGMENT_NODE;
-(Node.prototype as any).NOTATION_NODE = NodeType.NOTATION_NODE; // historical
+Node.prototype.ELEMENT_NODE = NodeType.ELEMENT_NODE;
+Node.prototype.ATTRIBUTE_NODE = NodeType.ATTRIBUTE_NODE;
+Node.prototype.TEXT_NODE = NodeType.TEXT_NODE;
+Node.prototype.CDATA_SECTION_NODE = NodeType.CDATA_SECTION_NODE;
+Node.prototype.ENTITY_REFERENCE_NODE = NodeType.ENTITY_REFERENCE_NODE; // historical
+Node.prototype.ENTITY_NODE = NodeType.ENTITY_NODE; // historical
+Node.prototype.PROCESSING_INSTRUCTION_NODE = NodeType.PROCESSING_INSTRUCTION_NODE;
+Node.prototype.COMMENT_NODE = NodeType.COMMENT_NODE;
+Node.prototype.DOCUMENT_NODE = NodeType.DOCUMENT_NODE;
+Node.prototype.DOCUMENT_TYPE_NODE = NodeType.DOCUMENT_TYPE_NODE;
+Node.prototype.DOCUMENT_FRAGMENT_NODE = NodeType.DOCUMENT_FRAGMENT_NODE;
+Node.prototype.NOTATION_NODE = NodeType.NOTATION_NODE; // historical

--- a/test/NodeType.tests.ts
+++ b/test/NodeType.tests.ts
@@ -1,0 +1,34 @@
+import * as slimdom from '../src/index';
+
+describe('NodeType', () => {
+	it('is exposed as constants on the Node constructor', () => {
+		expect(slimdom.Node.ELEMENT_NODE).toBe(1);
+		expect(slimdom.Node.ATTRIBUTE_NODE).toBe(2);
+		expect(slimdom.Node.TEXT_NODE).toBe(3);
+		expect(slimdom.Node.CDATA_SECTION_NODE).toBe(4);
+		expect(slimdom.Node.ENTITY_REFERENCE_NODE).toBe(5);
+		expect(slimdom.Node.ENTITY_NODE).toBe(6);
+		expect(slimdom.Node.PROCESSING_INSTRUCTION_NODE).toBe(7);
+		expect(slimdom.Node.COMMENT_NODE).toBe(8);
+		expect(slimdom.Node.DOCUMENT_NODE).toBe(9);
+		expect(slimdom.Node.DOCUMENT_TYPE_NODE).toBe(10);
+		expect(slimdom.Node.DOCUMENT_FRAGMENT_NODE).toBe(11);
+		expect(slimdom.Node.NOTATION_NODE).toBe(12);
+	});
+
+	it('is exposed as properties on Node instances', () => {
+		const node = slimdom.document.createElement('test');
+		expect(node.ELEMENT_NODE).toBe(1);
+		expect(node.ATTRIBUTE_NODE).toBe(2);
+		expect(node.TEXT_NODE).toBe(3);
+		expect(node.CDATA_SECTION_NODE).toBe(4);
+		expect(node.ENTITY_REFERENCE_NODE).toBe(5);
+		expect(node.ENTITY_NODE).toBe(6);
+		expect(node.PROCESSING_INSTRUCTION_NODE).toBe(7);
+		expect(node.COMMENT_NODE).toBe(8);
+		expect(node.DOCUMENT_NODE).toBe(9);
+		expect(node.DOCUMENT_TYPE_NODE).toBe(10);
+		expect(node.DOCUMENT_FRAGMENT_NODE).toBe(11);
+		expect(node.NOTATION_NODE).toBe(12);
+	});
+});

